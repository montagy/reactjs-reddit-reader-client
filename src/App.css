.wrapper {
  min-height: 100vh;
  background-color: color(var(--baseColor) b(+10%));
  display: flex;
  position: relative;
  z-index: 1;
}
:root {
  --primary: #108ee9;
}
.container {
  width: 100%;
  display: flex;
  justify-content: space-between;
  position: relative;
  transition: margin-left .3s ease-in-out;
  & section {
    flex-grow: 1;
  }
  &:before {
    content: "";
    display: block;
    height: 200px;
    width: 100%;
    background-image: linear-gradient(to left, #19111f 0%, #2c1e37 100%);
    position: absolute;
    z-index: -1;
    top: 0;
    left: 0;
    right: 0;
  }
}
.toggle {
  display: block;
  font-size: 1em;
  padding: 0 .5em;
  width: 4em;
  height: @width;
  cursor: pointer;
  border: 1px solid var(--primary);
  border-radius: 0.25em;
  transition: all .3s ease-in-out;
  background-color: var(--primary);
  & > span {
    display: block;
    min-height: 5px;
    width: 80%;
    margin: 0.5em auto;
    background-color: #FFF;
    transition: transform .5s ease, opacity .5s, background-color .5s;
  }
  &:hover {
    background-color: color(var(--primary) l(+10%));
    border-color: @background-color;
  }
}
.firstBar {
  transform: translateY(9px) rotate(45deg);
}
.lastBar {
  transform: translateY(-9px) rotate(-45deg);
}

