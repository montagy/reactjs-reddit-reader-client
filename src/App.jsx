import React from 'react';
import { HashRouter, Route } from 'react-router-dom';
import { Home, SubReddit, Post } from './pages';
import Pannel from './molecules/Pannel';
import global from './global';
import styles from './App.css';

class App extends React.Component {
  state = {
    showPannel: false,
  };
  togglePannel = e => {
    e.preventDefault();
    this.setState(prevState => ({
      showPannel: !prevState.showPannel,
    }));
  };
  render() {
    return (
      <HashRouter>
        <div className={styles.wrapper}>
          <Pannel style={{ width: this.state.showPannel ? '30%' : '0' }} />
          <div
            className={styles.container}
            style={{ marginLeft: this.state.showPannel ? '30%' : '0' }}
          >
            <button onClick={this.togglePannel} className={styles.toggle}>
              <span className={this.state.showPannel ? styles.firstBar : ''} />
              <span
                style={{ opacity: this.state.showPannel ? '0' : '1' }}
              />
              <span className={this.state.showPannel ? styles.lastBar : ''} />
            </button>
            <Route exact path="/" component={Home} />
            <Route path="/r/:sub" component={SubReddit} />
            <Route path="/p/:permalink" component={Post} />
          </div>
        </div>
      </HashRouter>
    );
  }
}

export default App;
