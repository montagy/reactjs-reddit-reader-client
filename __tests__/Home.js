import React from 'react';
import renderer from 'react-test-renderer';
import App from '../src/App.jsx';
describe('Home', () => {
  beforeEach(() => {
    global.localStorage = jest.fn().mockReturnValue({
      getItem: jest.fn(),
      setItem: jest.fn(),
    });
    global.fetch = jest.fn().mockImplementation(url => {
      return new Promise((resolve, reject) => {
        resolve({
          json: function() {
            return {
              url,
              id: 1,
              ok: true,
            };
          },
        });
      });
    });
  })
  it('snapshot', () => {
    const tree = renderer
      .create(
        <App />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
