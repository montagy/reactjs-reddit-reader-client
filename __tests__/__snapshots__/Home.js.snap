// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="pannel"
    style={
      Object {
        "width": "0",
      }
    }
  >
    <div>
      <form>
        <input
          placeholder="reddit"
          type="text"
        />
        <button>
          -&gt;
        </button>
      </form>
      <div>
        Something
      </div>
      <div>
        Something
      </div>
      <div>
        Something
      </div>
      <div>
        Something
      </div>
    </div>
  </div>
  <div
    className="container"
    style={
      Object {
        "marginLeft": "0",
      }
    }
  >
    <button
      className="toggle"
      onClick={[Function]}
    >
      <span
        className=""
      />
      <span
        style={
          Object {
            "opacity": "1",
          }
        }
      />
      <span
        className=""
      />
    </button>
    <section
      className="wrapper"
    >
      <div>
        <h1>
          Home
        </h1>
      </div>
      <main>
        <div
          className="loading"
        />
      </main>
    </section>
  </div>
</div>
`;
